### ISSUE TITLE
Changeset.unique_constraint not working with MySQL 8.0.19

### ISSUE THREAD

**Issue by kuffel** [elixir-ecto/ecto_sql#175]
The name for the unique index is not correctly extracted from the changeset and creates errors when running the tests against a MySQL 8.0.19 server.

### Environment

* Elixir version (elixir -v): Elixir 1.9.1 (compiled with Erlang/OTP 20)
* Database and version: MySQL Server 8.0.19
* Ecto version (mix deps): 3.3.0
* Database adapter and version (mix deps): myxql 0.3.0
* Operating system: Ubuntu 18.04

### Current behavior

When you run the ecto_sql tests against MySQL 8.0.18 everything is fine and all tests pass.

```
git clone https://github.com/elixir-ecto/ecto_sql.git
cd ecto_sql
docker run -d --name db -e MYSQL_ROOT_PASSWORD='' -e MYSQL_ALLOW_EMPTY_PASSWORD='true' -p 3306:3306 -t mysql:8.0.18
ECTO_ADAPTER=myxql mix test
``` 
When you run the ecto_sql tests against MySQL 8.0.19, 5 tests around the unique_constraint fail.

```
docker stop db && docker rm db
docker run -d --name db -e MYSQL_ROOT_PASSWORD='' -e MYSQL_ALLOW_EMPTY_PASSWORD='true' -p 3306:3306 -t mysql:8.0.19
ECTO_ADAPTER=myxql mix test
``` 
The failure is due to a different index name. If you explicitly use the name option it works.

```
1) test unique constraint from association (Ecto.Integration.RepoTest)
     deps/ecto/integration_test/cases/repo.exs:360
     ** (Ecto.ConstraintError) constraint error when attempting to insert struct:
     
         * posts.posts_uuid_index (unique_constraint)
     
     If you would like to stop this constraint violation from raising an
     exception and instead add it as an error to your changeset, please
     call `unique_constraint/3` on your changeset with the constraint
     `:name` as an option.
     
     The changeset defined the following constraints:
     
         * posts_uuid_index (unique_constraint)
     
     code: TestRepo.insert %User{
     stacktrace:
       (ecto) lib/ecto/repo/schema.ex:694: anonymous fn/4 in Ecto.Repo.Schema.constraints_to_errors/3
       (elixir) lib/enum.ex:1336: Enum."-map/2-lists^map/1-0-"/2
       (ecto) lib/ecto/repo/schema.ex:679: Ecto.Repo.Schema.constraints_to_errors/3
       (ecto) lib/ecto/repo/schema.ex:274: anonymous fn/15 in Ecto.Repo.Schema.do_insert/4
       (ecto) lib/ecto/association.ex:680: Ecto.Association.Has.on_repo_change/5
       (ecto) lib/ecto/association.ex:447: anonymous fn/8 in Ecto.Association.on_repo_change/7
       (elixir) lib/enum.ex:1948: Enum."-reduce/3-lists^foldl/2-0-"/3
       (ecto) lib/ecto/association.ex:443: Ecto.Association.on_repo_change/7
       (elixir) lib/enum.ex:1948: Enum."-reduce/3-lists^foldl/2-0-"/3
       (ecto) lib/ecto/association.ex:407: Ecto.Association.on_repo_change/4
       (ecto) lib/ecto/repo/schema.ex:842: Ecto.Repo.Schema.process_children/5
       (ecto) lib/ecto/repo/schema.ex:919: anonymous fn/3 in Ecto.Repo.Schema.wrap_in_transaction/6
       (ecto_sql) lib/ecto/adapters/sql.ex:903: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
       (db_connection) lib/db_connection.ex:1427: DBConnection.run_transaction/4
       deps/ecto/integration_test/cases/repo.exs:365: (test)
```

**Comment by wojtekmach** [comment]
Thank you for the report. Looks like mysql started to include table name in the constraint violation error and we would need to handle that.

**Comment by wojtekmach** [comment]
Until we have a solution for this as a workaround folks can do this in their app code:


```diff
-  |> unique_constraint(:uuid)
+  |> unique_constraint(:uuid, name: "posts.posts_uuid_index")
```

but yes, we should solve this properly in ecto_sql.

**Comment by wojtekmach** [comment]
Fixed in #178.
